Add YAML configuration code samples

blazember opened the pull request.

leszko commented on learning-basics/configure-yaml/README.md:
```diff
@@ -0,0 +1,43 @@
+  
+The sample `YamlConfigClasspath` loads the member configuration `hazelcast-sample.yaml` from the classpath and 
+demonstrates variable replacement in YAML configuration by defining the member's port in variable taken from 
+the system properties and then from a provided `Properties` instance.   
```
Would it make sense to add the information on how to run this sample and what the reader should observe?

blazember replied:
The samples are implemented in main methods without using shell scripts. I can add that, but I think it's not a question for a java engineer.

What the user should observe when running the samples is added as inline comments.

leszko commented on learning-basics/configure-yaml/src/main/java/ConfigLookup.java:
```diff
@@ -0,0 +1,32 @@
+        // XML takes precedence: loading member configuration hazelcast.xml from the classpath
+        // despite the presence of hazelcast.yaml
+        HazelcastInstance member = Hazelcast.newHazelcastInstance();
+        sleepMillis(1000);
```
Why do you need this sleeps? For the user to observe the results? Or this is necessary for some reason? Could you maybe comment it in the code?

blazember replied:
I used the `configuration-xml` sample as a basis, the sleeps were there. Removed, it wasn't necessary.

leszko commented on learning-basics/configure-yaml/src/main/resources/hazelcast-client.yaml:
```diff
@@ -0,0 +1,6 @@
+hazelcast-client:
+  import:
```
I guess this import is for the reader to observe something, would it make sense to describe it in README?

blazember replied:
Added to the README that the first test demonstrates composing YAML configuration files.

leszko commented on learning-basics/configure-yaml/src/main/resources/hazelcast-client.yaml:
```diff
@@ -0,0 +1,6 @@
+  import:
+    - hazelcast-client-network-config.yaml
+  group:
+    name: dev-configured-from-yaml
```
Do you use it somehow in the code sample?

blazember replied:
It is needed to be the same in member and client config. It's better to be unique in the samples so no members and clients join from or to instances started by other samples by mistake.

blazember commented:
Thanks for the comments @leszko, addressed them. Could you please take a second look?

blazember commented:
Thanks for the review @lazerion @leszko :+1: 

Squashed and added a new section referencing the `hazelcast-full-example.yaml` configuration to the README.
